src/tribler-core/tribler_core/modules/bandwidth_accounting/database.py

```diff
@@ -0,0 +1,122 @@
+from pathlib import Path
+from typing import List
+
+from pony.orm import Database, count, db_session, select, sum
+
+from tribler_core.modules.bandwidth_accounting import history, misc, transaction
+from tribler_core.modules.bandwidth_accounting.transaction import BandwidthTransactionData
+
+
+class BandwidthDatabase:
+    """
+    Simple database that stores bandwidth transactions in Tribler as a work graph.
+    """
+    CURRENT_DB_VERSION = 8
+    MAX_HISTORY_ITEMS = 100  # The maximum number of history items to store.
+
+    def __init__(self, db_path: Path, my_pub_key: bytes) -> None:
+        """
+        Sets up the persistence layer ready for use.
+        :param db_path: The full path of the database.
+        :param my_pub_key: The public key of the user operating the database.
+        """
+        self.db_path = db_path
+        self.my_pub_key = my_pub_key
+        create_db = str(db_path) == ":memory:" or not self.db_path.is_file()
+        self.database = Database()
+
+        self.MiscData = misc.define_binding(self.database)
+        self.BandwidthTransaction = transaction.define_binding(self)
+        self.BandwidthHistory = history.define_binding(self)
+
+        self.database.bind(provider='sqlite', filename=str(db_path), create_db=str(create_db), timeout=120.0)
```

Comment:
`create_db` option is a bool flag, conversion to `str` is probably error?

Reply:
Good catch! I copied this particular line from the metadata store that was already there, so I guess it would also be a bug there?

Reply:
Yep! Definitely a bug in my code :clap: 
That's the most prominent vector of bug proliferation - blind copy-paste.